pfhedge/nn/functional.py

```diff
@@ -518,7 +518,19 @@ def d1(log_moneyness: Tensor, time_to_maturity: Tensor, volatility: Tensor) -> T
         torch.Tensor
     """
     s, t, v = broadcast_all(log_moneyness, time_to_maturity, volatility)
-    return (s + (v.square() / 2) * t).div(v * t.sqrt())
+    if not (t >= 0).all():
+        raise ValueError("all elements in time_to_maturity have to be non-negative")
+    if not (v >= 0).all():
+        raise ValueError("all elements in volatility have to be non-negative")
+    numerator = s + (v.square() / 2) * t
+    denominator = v * t.sqrt()
+    zero_div_zero_mask = (numerator == 0).logical_and(denominator == 0)
+    denominator = torch.where(
+        zero_div_zero_mask,
+        torch.zeros_like(denominator) + torch.finfo().tiny,
+        denominator,
+    )
+    return numerator.div(denominator)
```

Comment:
Nice that nonzero/0 is inf rather than a large but finite number (expected behavior right?)

NIT: Possibly it could be a bit more intuitive to express it like:

output = numerator / denominator
output.where(numerator == 0 and denominator == 0, 0)

Reply:
> Nice that nonzero/0 is inf rather than a large but finite number (expected behavior right?)

Yes, I meant to.

According to your suggestion, I'll fix